Orange/widgets/utils/itemmodels.py

```diff
@@ -490,13 +491,14 @@ def __setitem__(self, s, value):
                 value = list(value)
             separators = [start + i for i, v in enumerate(value) if v is self.Separator]
             self.beginInsertRows(QModelIndex(), start, start + len(value) - 1)
-            self._list[s] = value
```

Comment:
The 'items' corresponding to the slice which should have been overwritten were already removed 5 lines before.